src/coprocessor/dag/expr/builtin_like.rs

```diff
@@ -0,0 +1,241 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::i64;
+use std::slice::Iter;
+use std::str;
+
+use super::{EvalContext, FnCall, Result};
+use coprocessor::codec::Datum;
+use regex::Regex;
+
+const MAX_RECURSE_LEVEL: usize = 1024;
+
+impl FnCall {
+    /// NOTE: LIKE compare target with pattern as bytes, even if they have different
+    /// charsets. This behaviour is for keeping compatible with TiDB. But MySQL
+    /// compare them as bytes only if any charset of pattern or target is binary,
+    /// otherwise MySQL will compare decoded string.
+    pub fn like(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        let target = try_opt!(self.children[0].eval_string(ctx, row));
+        let pattern = try_opt!(self.children[1].eval_string(ctx, row));
+        let escape = try_opt!(self.children[2].eval_int(ctx, row)) as u32;
+        Ok(Some(like(&target, &pattern, escape, 0)? as i64))
+    }
+
+    pub fn regexp(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        let target = try_opt!(self.children[0].eval_string(ctx, row));
+        let pattern = try_opt!(self.children[1].eval_string(ctx, row));
+
+        let mut p = String::from("(?i)");
```

Comment:
why? Seems tidb does not have the corresponding step: https://github.com/pingcap/tidb/blob/master/expression/builtin_like.go#L128

Reply:
The corresponding step is in 'builtinRegexpSig'(https://github.com/pingcap/tidb/blob/5022b5b2cc31ef64b4115482aaf44a50a6618024/expression/builtin_like.go#L159) which makes Regexp() case-insensitive